baselines/FedVSSL/README.md

```diff
@@ -0,0 +1,144 @@
+* After pretraining it finetunes the R3D-18 model on UCF-101 dataset.
+
+**Dataset:** This baselines only include pre-training and fine-tuning with UCF-101 dataset. However, we also provide the script files to generate the partitions for Kinetics-400 datasets. 
+For UCF-101 dataset, one can simply run the `dataset_preparation.py` file to download and generate the iid splits for UCF-101 datasets.
+
+| Dataset | #classes | #partitions | partitioning method |  partition settings  |
+|:--------|:--------:|:-----------:| :---: |:--------------------:|
+| UCF101  |   101    |     10      | randomly partitioned |       uniform        |
+| Kinectics-400    |   400    |     100     | randomly partitioned | 8 classes per client |
+
+**Training Hyperparameters:** The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run python main.py directly)
+
+| Description        |            Default Value            |
+|:-------------------|:-----------------------------------:|
+| total clients      |                 2                  |
+| clients per round  |                 2                  | 
+| number of rounds	  |                 20                  | 
+| client resources	  | {'num_cpus': 2.0, 'num_gpus': 1.0 } | 
+| optimizer	  |                 SGD                  | 
+| alpha coefficient	 |                 0.9                 | 
+| beta coefficient	  |                  1                  | 
+
+## Environment Setup
+To construct the Python environment follow these steps:
+
+```bash
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+
+# install PyTorch with GPU support.
+pip install torch==1.7.1+cu110 torchvision==0.8.2+cu110 torchaudio==0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
+
+# install mmcv package
+pip install mmcv==1.2.4
+```
+
+## Running the Experiments
+To run FedVSSL with UCF-101 baseline, first follow the instruction in `dataset_preparation.py` to download the dataset, complete pre-processing and data partitioning for FL.
+Then, ensure you have activated your Poetry environment (execute `poetry shell` from this directory), then:
+
+```bash
+# direct to FedVSSL code directory
+cd FedVSSL
+
+# clone CtP repo
+git clone https://github.com/yan-gao-GY/CtP.git
+
+# run federated SSL training with FedVSSL
+python -m main --pre_training=True # this will run using the default settings.
+
+# you can override settings directly from the command line
+python -m main --pre_training=True --mix_coeff=1 --rounds=100 # will set hyper-parameter alpha to 1 and the number of rounds to 100
+
+# run downstream fine-tuning with pre-trained SSL model
+python -m main --pre_training=False # this will run using the default settings.
+```
+
+To run using FedAvg:
+```bash
+# this will run FedAvg baseline
+# This is done so to match the experimental setup in the paper
+python -m main --fedavg=True
```

Comment:
The code should be run as `python -m <baseline_package_name>.main ...`

Reply:
This will require to adjust the imports in several of your python files. For instance in you `main.py` instead of `from strategy import FedVSSL` you'll do `from <package_name>.strategy import FedVSSL`

Reply:
As i mentioned earlier, please rename your baseline directory to `fedvssl`, in that way your package name will be `fedvssl`.